go/pkg/tool/tool.go

```diff
@@ -216,6 +217,21 @@ func (c *Client) DownloadBlob(ctx context.Context, blobDigest, path string) (str
 	return string(contents), nil
 }
 
+// UploadBlob uploads a blob from the specified path into the remote cache.
+func (c *Client) UploadBlob(ctx context.Context, blobDigest, path string) error {
+	dg, err := digest.NewFromString(blobDigest)
```

Comment:
Use NewFromFile here to compute the digest from the path.

Reply:
done